control_plane/src/storage.rs

```diff
@@ -515,22 +506,27 @@ impl PageServerNode {
         new_timeline_id: Option<ZTimelineId>,
         ancestor_start_lsn: Option<Lsn>,
         ancestor_timeline_id: Option<ZTimelineId>,
-    ) -> anyhow::Result<Option<TimelineInfo>> {
-        let timeline_info_response = self
-            .http_request(
-                Method::POST,
-                format!("{}/tenant/{}/timeline", self.http_base_url, tenant_id),
+    ) -> anyhow::Result<TimelineInfo> {
```

Comment:
Why is this change needed?

Reply:
`Option<TimelineInfo>` is `None` according to our logic only if we fail to create a timeline:
https://github.com/neondatabase/neon/blob/e593cbaabafafb6f54c9ddcd5d1d9f04d1bd4490/pageserver/src/http/routes.rs#L244

So instead of doing various `.unwrap()` or other error handling above the client, I've simplified it to validating the presence here.